modules/cudacodec/src/frame_queue.hpp

```diff
@@ -50,11 +50,16 @@
 class RawPacket {
 public:
     RawPacket(const unsigned char* _data, const size_t _size = 0, const bool _containsKeyFrame = false);
-    unsigned char* Data() const { return *data; }
-    size_t size;
-    bool containsKeyFrame;
+    ~RawPacket();
+    RawPacket(const RawPacket& other);
+    RawPacket(RawPacket&& other) noexcept;
+    RawPacket& operator=(const RawPacket& other);
+    RawPacket& operator=(RawPacket&& other) noexcept;
+    unsigned char* Data() const { return data; }
+    size_t size = 0;
+    bool containsKeyFrame = false;
 private:
-    cv::Ptr<unsigned char*> data = 0;
+    unsigned char* data = 0;
```

Comment:
Another option. Make `data` `std::vector<unsigned char>`.
Imho it's preferable:
[Godbolt link](https://godbolt.org/z/MvsMKMPnY). Everything can be defaulted.

```cpp
#include <iostream>
#include <vector>

std::ostream& print(std::ostream& stream, const unsigned char* data, size_t size) {
    stream << "[";
    if (size > 0 && data != nullptr) {
        stream << static_cast<int>(*data);
        for (size_t i = 1; i < size; ++i) {
            stream << ", " << static_cast<int>(data[i]);
        }
    }
    return stream << "]";
}

class RawPacket {
public:
    RawPacket(const unsigned char* data, size_t size) : data_{data, data + size} {
        std::cout << this << " from ";
        print(std::cout, data, size) << "\n";
    }

    RawPacket(const RawPacket& other) = default;

    RawPacket& operator=(const RawPacket& other) = default;

    RawPacket(RawPacket&& other) noexcept = default;

    RawPacket& operator=(RawPacket&& other) noexcept = default;

    ~RawPacket() = default;

    size_t size() const noexcept {
        return data_.size();
    }

    const unsigned char* data() const noexcept {
        return data_.data();
    }

private:
    std::vector<unsigned char> data_;
};

std::ostream& operator<<(std::ostream& stream, const RawPacket& packet) {
    stream << &packet << " contains ";
    return print(stream, packet.data(), packet.size());
}

int main() {
    const unsigned char source[] {1, 2, 3, 4, 5, 6, 7, 8};
    RawPacket packet(source, 3);
    std::cout << packet << std::endl;
    RawPacket packet2 = std::move(packet);
    std::cout << packet2 << std::endl;

    RawPacket packet3 = packet2;
    std::cout << packet3 << std::endl;
    RawPacket packet4(source + 2, 6);
    std::cout << packet4 << std::endl;
    packet4 = packet2;
    std::cout << packet4 << std::endl;
    return 0;
}
```

Reply:
Thanks for the detailed review, I agree vector would be better than mixing smart pointers, I'll address this when I am back.